File: src/main/java/org/voegtle/weatherwidget/util/ContextUtil.java
package org.voegtle.weatherwidget.util;

import android.content.Context;
import android.content.pm.PackageManager;
import android.util.Log;
import org.voegtle.weatherwidget.widget.WidgetUpdateTask;

public class ContextUtil {

  public static Integer getBuildNumber(Context context) {
    Integer buildNumber = null;
    try {
      PackageManager packageManager = context.getPackageManager();
      String packageName = context.getPackageName();
      buildNumber = packageManager.getPackageInfo(packageName, 0).versionCode;
    } catch (PackageManager.NameNotFoundException ex) {
      Log.e(ContextUtil.class.toString(), "Failed to retrieve build number", ex);
    }
    return buildNumber;
  }

  public static String getVersion(Context context) {
    String version = null;
    try {
      PackageManager packageManager = context.getPackageManager();
      String packageName = context.getPackageName();
      version = packageManager.getPackageInfo(packageName, 0).versionName;
    } catch (PackageManager.NameNotFoundException ex) {
      Log.e(ContextUtil.class.toString(), "Failed to retrieve app version", ex);
    }
    return version;
  }

}

File: src/main/java/org/voegtle/weatherwidget/util/ContextUtil.kt
package org.voegtle.weatherwidget.util

import android.content.Context
import android.content.pm.PackageManager
import android.util.Log
import org.voegtle.weatherwidget.widget.WidgetUpdateTask

object ContextUtil {

  fun getBuildNumber(context: Context): Int {
    var buildNumber: Int = -1
    try {
      val packageManager = context.packageManager
      val packageName = context.packageName
      buildNumber = packageManager.getPackageInfo(packageName, 0).versionCode
    } catch (ex: PackageManager.NameNotFoundException) {
      Log.e(ContextUtil::class.java.toString(), "Failed to retrieve build number", ex)
    }

    return buildNumber
  }

  fun getVersion(context: Context): String {
    var version: String = "unknown"
    try {
      val packageManager = context.packageManager
      val packageName = context.packageName
      version = packageManager.getPackageInfo(packageName, 0).versionName
    } catch (ex: PackageManager.NameNotFoundException) {
      Log.e(ContextUtil::class.java.toString(), "Failed to retrieve app version", ex)
    }

    return version
  }

}
